libs/akita/src/lib/trackIdChanges.ts

```diff
@@ -0,0 +1,57 @@
+import { EntityState, EntityStore, getEntityType, getIDType, ID, QueryEntity } from '@datorama/akita';
+import { merge, MonoTypeOperatorFunction, Observable, of, Operator, Subscriber, TeardownLogic, UnaryFunction } from 'rxjs';
+import { distinctUntilChanged, filter, first, switchMap, tap } from 'rxjs/operators';
+import { OperatorFunction } from 'rxjs/src/internal/types';
+
+type getEntityStoreState<T extends EntityStore<any>> = T extends EntityStore<infer S> ? S : never;
+
+/**
+ * Track id updates of an entity and re-evaluation the query with the changed entity id.
+ * Hint: Don't place the operator after other operators in the same pipeline as those will be skipped on
+ * re-evaluation.
+ * @param store The entity store in which the entity is located.
+ * @example
+ *
+ *   query.selectEntity(1).pipe(trackIdChanges(store)).subscribe(entity => { ... })
+ *
+ */
+export function trackIdChanges<K extends EntityStore<S, T>, S extends EntityState<T> = getEntityStoreState<K>, T = getEntityType<S>>(store: K): MonoTypeOperatorFunction<T> {
+  return (source) => source.lift<T>(new TrackIdChanges(store));
+}
+
+class TrackIdChanges<K extends EntityStore<S, T>, S extends EntityState<T>, T = getEntityType<S>> implements Operator<T, T> {
+  readonly query: QueryEntity<S, T>;
+
+  constructor(readonly store: K) {
+    this.query = new QueryEntity((store as unknown) as EntityStore<S>);
+  }
+
+  call(subscriber: Subscriber<T>, source: Observable<T>): TeardownLogic {
+    return source
+      .pipe(
+        first(),
+        switchMap((entity) => {
+          let currId = entity[this.store.config.idKey];
+          let select = this.query.selectEntity(currId);
```

Comment:
Where do you use the `select` variable? The only place I see is overriding it.

Reply:
I missed to remove it. The variable become redundant after using the `pending` state.